Tests/Test 5/wave_game.cpp

```diff
@@ -0,0 +1,65 @@
+#include <iostream>
+#include<queue>
+using namespace std;
+class Node{
+    public:
+    int x; // x coordainate
+    int y; // y coordainate
+    int dist; // distance from the start
+    Node(int x, int y, int distance){ //constructor
+        this->x=x;
+        this->y=y;
+        this->dist=distance;
+    }
+};
+int arr[500][500],n,m;
+int startX,startY,Distance; // the start coordinates and the distance
+bool used[500][500]; // boolean matrix where we will save if we used curtain vertex 
+long long int sum; // the sum of all vertices at the given distance from start
```

Comment:
Do you actually need so big number? from the constraints of the task simple int should be enough.